src/errors/kinds.rs

```diff
@@ -118,6 +102,16 @@ pub enum ErrorKind {
     LiteralSingleError,
     #[strum(serialize = "literal_error", message = "Value must be one of: {expected}")]
     LiteralMultipleError,
+    // ---------------------
+    // comparison errors
```

Comment:
What I was considering, in addition to the changes present, is whether min/max length errors should get aggregated?
This would require passing the involved Py type's name or removing this information from the error message.
@samuelcolvin thoughts?

Reply:
Not quite sure what you mean?

Reply:
Apologies, I made this sentence unnecessarily unreadable.
What I meant is I wondered whether, say, DictTooShort and ListTooShort errors should become one, too, just like the comparison errors did in this very PR.

Reply:
Yes I guess so, but it'll require a context variable for "items" vs. "characters" etc.

Note, I'll also started some of these changes in #77.

Reply:
Oh, alright, I didn't realise. I guess we can close this one then :D